### DISCUSSION TITLE
Convert RGB png to CMYK tiff

### DISCUSSION THREAD

**Discussion by Zelfapp** [ImageMagick/ImageMagick#2465] [Help] [answered]
How do I keep transparency and correctly convert a RGB PNG to CMYK tiff?

I've been trying the following to no avail. I'm adding text using `label` to the attached png image and then trying to convert to CMYK.

I have looked here at [profiles](http://www.imagemagick.org/Usage/formats/#profiles), but am not grasping what to do here. We need to use US Web Coated (SWOP) CMYK profile.

```
convert 2nd-amendment-american-flag.png \
-resize 3300x4220 \
-background transparent \
-font Army-Wide \
-fill "#000000" \
-gravity Center \
-interline-spacing 65 \
-pointsize 180 \
label:"Text Line One\nText Line Two" \
-smush 180 \
-trim \
-colorspace cmyk \
result.tiff
```

![2nd-amendment-american-flag](https://user-images.githubusercontent.com/11559805/91220930-28ccf080-e6da-11ea-8c4b-e3580a435562.png)

**Comment by snibgo** [comment, marked as answer]
I assume your input is encoded as sRGB.

I suggest you use profiles, not "-colorspace CMYK".

First, use "-profile sRGB.icc" which will either assign that profile, if it doesn't already have one, or convert it if it has. Then use "-profile USWebCoatedSWOP.icc" or any other CMYK profile, to convert it to that profile.

Put these operations immediately before writing your output:
```
convert 2nd-amendment-american-flag.png \
:
:
-profile sRGB.icc -profile USWebCoatedSWOP.icc \
result.tiff
```
If the ICC files are not in your current directory, provide the full paths. ICC profiles are available from http://www.color.org/registry/index.xalter and many other places.
